### DISCUSSION TITLE
Integration to Home Assistant

### DISCUSSION THREAD

**Discussion by oldgitdaddy** [tillsteinbach/VWsFriend#89] [Ideas]
To boost adoption and to also make use of the home assistant community, did anybody already make VWsFriend available in home assistant? Would be great to see this appearing in the future as an integration in Home Assistant.

**Comment by tillsteinbach** [comment]
what would you expect from such an integration? If it is just about getting the we connect data in Home Assistant there should be a solution.
Home Assistant can speak MQTT: https://www.home-assistant.io/integrations/mqtt/
I have made a MQTT client for WeConnect: https://github.com/tillsteinbach/WeConnect-mqtt
This should enable you to get data into home assistant. But you would not hav additional data such as the trips or charging sessions.

**Comment by oldgitdaddy** [comment]
In most cases users have Home Assistant OS installed, which is a pre-packaged bundle of operating system and home assistant. Home Assistant OS does not allow to install further docker images out-of-the-box on the same machine. Further, users may only have one central server instance to manage their smart home (e.g. raspberry pi). So such an integration would basically help to address this need, because otherwise users would just not consider to using VWsFriend. Thank you for further considering.

**Reply by tillsteinbach** [reply]
Ah, understood. Would be great to find a Home Assistant user to help here. I don't have Home Assistant in use right now.

**Reply by northalpha** [reply]
Ready to help, i hav an ID3 and a fully fledged Home Assistant Setup at home und would be more than happy to assist here.

**Comment by gapthorpe** [comment]
Appreciate this is an old thread, but I got an Id.4 yesterday and have been playing with this tool (amazing work and thanks!!)
I have configured mqtt to try and get the data into HomeAssistant, but as there is no discovery message in the required format it doesn’t appear. Just wondered if anyone had managed to get into HA and any advice?  Thanks!

**Reply by tillsteinbach** [reply]
There is a homeassistant integration that was built ontop of m library: https://github.com/mitch-dc/volkswagen_we_connect_id

**Reply by gapthorpe** [reply]
Thanks - I have taken a look at that also, but was hoping that I could use mqtt so that I can use your awesome grafana dashboards but not need to b polling the vw servers twice

**Comment by RoMaTiX99** [comment]
Hi @tillsteinbach, 

first of all: thank you for this awesome project! 

I have my instance up and running flawlessly so far. However I'd also like to integrate it into my HomeAssistant Instance. I do not need the data (I use the official VW Integration already, since it's cloud-dependent either way). BUT I would love to have the Grafana Dashboards as a Panel oder Dashboard in my HomeAssistant. 

HomeAssistants offers the possibility, to show Websites inside Dashboards or Panel-Cards. This would be the easiest way to achieve this imho. You can either: 
A. Do Integration via the local ip 
B. Do Integration via public url (ssl encrypted) 

When you connect to your HomeAssistant via HTTPS (which should be really common), it only supports loading iframes from https sources. So basically you would NEED to pull the data via a ssl secured connection. 

For me I have set up everything so far that I can also access the VWsFriend GF Dashboard via Public URL, SSL encrypted. BUT I can't integrate it into my HomeAssistant. I guess this is because of the settings for the Grafana Database. These environmental parameters need to be changed: 
environment:
       - GF_AUTH_DISABLE_LOGIN_FORM=true
       - GF_AUTH_ANONYMOUS_ENABLED=true
       - GF_AUTH_ANONYMOUS_ORG_ROLE=Admin
       - GF_SECURITY_ALLOW_EMBEDDING=true

(a quick guide which settings need to be adjusted can be found here: https://thesmarthomejourney.com/2021/05/30/add-grafana-to-home-assistant/.)

My questions are: 
1. Is it possible to set these settings for the GF docker / installation? 
2. Are there any known alternatives for the Integration to HomeAssistant? 

Thanks a lot and best regards!
